tests/unit-tests/graphpartitioning/test_metadata.py

```diff
@@ -0,0 +1,731 @@
+            file_metadata,
+        )
+
+        # Create metadata.json file.
+        filename = os.path.join(root_dir, METADATA_NAME)
+        with open(filename, "w") as input_handle:
+            json.dump(input_dict, input_handle, indent=4)
+
+        # Test case here.
+        obj = MetadataSchema(METADATA_NAME, root_dir, init_maps=False)
+        obj._init_etype_maps()
+
+        # Check the initialization
+        try:
+            obj._init_etype_files()
+
+            if create_files:
+                for idx, etype in enumerate(etypes):
+                    (file_type, delimiter, data_files) = obj.get_etype_info(
+                        etype
+                    )
+                    assert file_type == edge_fmt, f"File type does not match."
+                    assert (
+                        delimiter == edge_fmt_del
+                    ), f"Delimiters does not match."
+                    assert (
+                        len(data_files) == idx + 1
+                    ), f"No. of edge files does not match."
+
+        except AssertionError as exp:
+            if create_files:
+                raise exp
+            else:
+                pass
+
+
+@pytest.mark.parametrize("ntypes", [[], [NODE1], [NODE1, NODE2]])
+@pytest.mark.parametrize("file_type", ["parquet", "numpy"])
+@pytest.mark.parametrize("delimiter", [" ", "/"])
+@pytest.mark.parametrize("create_files", [True, False])
+@pytest.mark.parametrize("prefix_dir", ["", "node_data"])
+@pytest.mark.parametrize("is_absolute", [True, False])
+def test_node_features(
+    ntypes, file_type, delimiter, create_files, prefix_dir, is_absolute
+):
+    """Unit tests when node features are present in the input graph."""
+
+    def run_test(root_dir, input_dict):
+        if root_dir is None:
+            root_dir = os.getcwd()
+
+        # Create metadata.json
+        filename = os.path.join(root_dir, METADATA_NAME)
+        with open(filename, "w") as input_handle:
+            json.dump(input_dict, input_handle, indent=4)
+
+        # Tests here.
+        obj = MetadataSchema(METADATA_NAME, root_dir, init_maps=False)
+        obj._init_ntype_maps()
+
+        try:
+            obj._init_ntype_feature_files()
+            if create_files:
+                for idx, ntype in enumerate(ntypes):
+                    (
+                        actual_file_type,
+                        actual_delimiter,
+                        data_files,
+                    ) = obj.get_ntype_feature_files(ntype, NFEAT1)
+                    assert (
+                        file_type == actual_file_type
+                    ), f"File type does not match."
+                    assert (
+                        delimiter == actual_delimiter
+                    ), f"Delimiters does not match."
+                    assert (
+                        len(data_files) == 5
+                    ), f"No. of node feature files does not match."
+
+        except AssertionError as exp:
```

Comment:
What is the purpose of this except?

Reply:
This file has been re-organized from scratch... mostly to address @thvasilo comments. Now this is outdated.